crates/solidity/outputs/npm/tests/src/cst-cursor.ts

```diff
@@ -0,0 +1,264 @@
+  expect(cursor.goToNext()).toBe(true);
+  expect(cursor.node.type).toBe(NodeType.Token);
+  expect(cursor.node).toBeInstanceOf(TokenNode);
+  expect((cursor.node as TokenNode).text).toBe(" ");
+  expect(cursor.goToNext()).toBe(true);
+  expect(cursor.node.type).toBe(NodeType.Token);
+  expect(cursor.node).toBeInstanceOf(TokenNode);
+  expect((cursor.node as TokenNode).text).toBe("2");
+  expect(cursor.goToNext()).toBe(true);
+  expect(cursor.node.type).toBe(NodeType.Token);
+  expect(cursor.node).toBeInstanceOf(TokenNode);
+  expect((cursor.node as TokenNode).text).toBe(";");
+  expect(cursor.goToNext()).toBe(false);
+});
+
+test("accessing the node in a cursor is slow", () => {
+  const source = "int256 constant z = 1 + 2;";
+  const language = new Language("0.8.1");
+
+  const { parseTree } = language.parse(ProductionKind.SourceUnit, source);
+
+  const cursor: Cursor = parseTree.cursor;
+  expect(cursor.isCompleted).toBe(false);
+
+  // This is slow.
+  let access_time = 0;
+  {
+    const start = performance.now();
+    for (let i = 0; i < 10; i++) {
+      do {
+        cursor.node;
+      } while (cursor.goToNext());
+      cursor.reset();
+    }
+    const end = performance.now();
+    access_time = end - start;
+  }
+
+  cursor.reset();
+
+  // This is faster
+  let no_access_time = 0;
+  {
+    const start = performance.now();
+    for (let i = 0; i < 10; i++) {
+      do {} while (cursor.goToNext());
+      cursor.reset();
+    }
+    const end = performance.now();
+    no_access_time = end - start;
+  }
+
+  const ratio = access_time / no_access_time;
+  console.log("cursor node access ratio", ratio);
+
+  // This is not good.
+  expect(ratio).toBeLessThan(10);
+});
+
+test("accessing the node in a cursor multiple times has a linear time cost", () => {
+  const source = "int256 constant z = 1 + 2;";
+  const language = new Language("0.8.1");
+
+  const { parseTree } = language.parse(ProductionKind.SourceUnit, source);
+
+  const cursor: Cursor = parseTree.cursor;
+
+  // This is slow.
+  let multiple_access_time = 0;
+  {
+    const start = performance.now();
+    for (let i = 0; i < 100; i++) {
+      do {
+        cursor.node;
+        cursor.node;
+        cursor.node;
+        cursor.node;
+      } while (cursor.goToNext());
+      cursor.reset();
+    }
+    const end = performance.now();
+    multiple_access_time = end - start;
+  }
+
+  cursor.reset();
+
+  // This is faster
+  let single_access_time = 0;
+  {
+    const start = performance.now();
+    for (let i = 0; i < 100; i++) {
+      do {
+        cursor.node;
+      } while (cursor.goToNext());
+      cursor.reset();
+    }
+    const end = performance.now();
+    single_access_time = end - start;
+  }
+
+  const ratio = multiple_access_time / single_access_time;
+  console.log("cursor node multiple access ratio", ratio);
+
+  // This is not good.
+  expect(ratio).toBeLessThan(4);
+});
+
+test("cursor find api is faster than testing the every node", () => {
```

Comment:
this test is still flakey unfortunately :( 

If you add any wrong/failed assertion in any other test, it will take a few milliseconds longer to render the error, slowing down the first one as well, and thus both will fail 😞 

I think testing timing this way in JS is not really consistent.  Not sure I understand what this test is maintaining/asserting. I wonder if there is a different way to achieve the same goal?